packages/jest-cli/src/search_source.js

```diff
@@ -238,4 +239,34 @@ export default class SearchSource {
       return Promise.resolve({tests: []});
     }
   }
+
+  async getTestPaths(
+    globalConfig: GlobalConfig,
+    changedFilesPromise: ?ChangedFilesPromise,
+  ): Promise<SearchResult> {
+    const searchResult = await this._getTestPaths(
+      globalConfig,
+      changedFilesPromise,
+    );
+
+    if (globalConfig.filter) {
+      const tests = searchResult.tests;
+
+      // $FlowFixMe: dynamic require.
+      const filter = require(globalConfig.filter);
+      const filtered = await filter(tests.map(test => test.path));
+
+      if (!Array.isArray(filtered)) {
+        throw new Error(
+          `Filter {globalConfig.filter} did not return a valid test list`
+        );
+      }
+
+      Object.assign({}, searchResult, {
```

Comment:
This is missing a return, lol.

Reply:
It misses the re-assignment, the `return` is below :)